### ISSUE TITLE
Finetunning Tibetan Text Layer Detection Model

### ISSUE THREAD

**Issue by karmatai** [OpenPecha/BEC#395]
## Description
In this card we going to train a model to detect different tibetan text layers(quotation,title,author etc) from tibetan text source by preparing dataset split from [tsadra data](https://github.com/OpenPecha-Data/catalog/blob/master/data/tsadra_repo_list.txt) , choose the best pretrain model,train and evaluate until we get production-ready model.

##Config

Parameter | Value
-- | --
Base checkpoint | jhu-clsp/mmBERT-base
Max sequence length | 8192 tokens
Stride | 5120 tokens
Learning rate | 1e-5
Effective batch size | 8 (2 × 4 accumulation steps)
Epochs | 10 (early stop patience 3)
Weight decay | 0.01
Grad clip | 0.3
Class weight cap | 5.0× most common layer
O class weight | 0.07× most common layer
Masked layers | BookTitle
Best model metric | soft_f1_tol1

## Subtask
- [x] Audit the training data ,remove mislabled data. Implement volume based 80/10/10 split.
- [x] Tokenize using mmbert tokenizer and give correct label tag.
- [x] Implement overlapping window slicing.
- [x] Training script to finetune mmbert for tibetan text layer detection.
- [x] Evaluate the result.
- [x] Prepare inference script.
- [x] Prepare reeaval script
- [x] Write experiment documentation and future implementation for improvment.
## Reviewer 
- [ ] @kaldan007

**Comment by karmatai** [comment]
After updating the decoding algorithm for evaluation the final result is :
layer	before	after	
Author	0.735	0.962	
Chapter	0.441	0.818	
Tsawa	0.104	0.395	
Sabche	0.199	0.394	
Yigchung	0.165	0.389	
Quote	0.102	0.373	
overall	0.134	0.398

**Comment by karmatai** [comment]
https://forum.openpecha.org/t/annotation-layer-detection-in-tibetan-texts-experiment-report/614
